tests/unit/test_orders.py

```diff
@@ -1,30 +1,61 @@
-from unittest.mock import patch
+from unittest.mock import Mock, patch
 
 import pytest
 
-import planet.clients.orders
-from planet.clients.orders import OrdersClient
-from planet.order_request import build_request, product
+from planet.clients.orders import OrdersClient, OrderError
 
 
 @pytest.mark.asyncio
 @patch("planet.clients.orders.ClientSession.post")
-async def test_create_order(mock_post, event_loop):
-    """Create an order."""
-    image_ids = ['3949357_1454705_2020-12-01_241c']
-    request = build_request('test_order',
-                            [product(image_ids, 'analytic', 'psorthotile')])
-
-    mock_post.return_value.__aenter__.return_value.json.return_value = {
-        "id": "lol",
-        "state": "wut"
+async def test_create_order_failure(mock_http_post):
+    """Get expected chained exception on fake service failure."""
+
+    # create_order() interacts with the Planet Orders service only via
+    # HTTP POST request. Here we patch the post() method of the aiohttp
+    # ClientSession class and program the mock object to raise an
+    # exception when its raise_for_status() method is called. Note that
+    # raise_for_status() is a normal (not-async) Python method and so
+    # we must explicitly use (not-async) Mock. Otherwise all the
+    # attributes of mock_http_post will be AsyncMock objects.
+    mock_http_post.return_value.__aenter__.return_value.raise_for_status = Mock(
+        side_effect=RuntimeError("Boom! Service failure"))
+
+    # In this demo version of OrdersClient a session is created
+    # internally and so we don't need to give it one.
+    client = OrdersClient(None)
+    with pytest.raises(OrderError) as excinfo:
+        await client.create_order({})
+
+    assert type(excinfo.value.__context__) == RuntimeError
+    assert str(excinfo.value.__context__) == "Boom! Service failure"
+
+
+@pytest.mark.asyncio
+@patch("planet.clients.orders.ClientSession.post")
+async def test_create_order_success(mock_http_post):
+    """Submit a fake order to fake service and get expected response."""
+
+    # In this test we want the response to the POST request to not raise
+    # an exception.
+    mock_http_post.return_value.__aenter__.return_value.raise_for_status = Mock(
+        return_value=None)
+
+    # create_order() async enters the response to an HTTP POST request
+    # and awaits the response's json() method. We program the return
+    # value of that method with an abbreviated Orders service response.
+    mock_http_post.return_value.__aenter__.return_value.json.return_value = {
+        "id": "lolwut",
+        "state": "queued"
     }
 
-    async def coro():
-        client = OrdersClient(None)
-        return await client.create_order(request)
+    client = OrdersClient(None)
+
+    # We provide an abbreviated fake orders request. create_orders()
+    # does not validate requests.
+    order = await client.create_order({"name": "fake order"})
 
-    order = await event_loop.create_task(coro())
```

Comment:
We don't need the extra coroutine or loop after all. pytest-asyncio takes care of this for us.